core/file_walker_test.go

```diff
@@ -12,6 +12,7 @@ import (
 
 type fakeFile struct {
 	fileName  string
+	fileDir   *File
```

Comment:
I don't think this is necessary, the fake file is representing "real" folder structure. It doesn't have to be consistent with `core.File`